runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/refresh/Heartbeat.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.dataflow.worker.windmill.work.refresh;
+
+import com.google.auto.value.AutoValue;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.runners.dataflow.worker.streaming.RefreshableWork;
+import org.apache.beam.runners.dataflow.worker.windmill.Windmill;
+
+/** Heartbeat requests and the work that was used to generate the heartbeat requests. */
+@AutoValue
+public abstract class Heartbeat {
```

Comment:
Seems like this shoudl be Heartbeats or HeartbeatsBatch , something to indicate it is multiple.

Can we change the colleciton of work and requests to be immutable? 
Since just created one place you could use some ImmutalbeMap builder there and change create to take those immutable maps.

Or could add some builder class for this that has immutable map builder inside of it.

Reply:
done